src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/VirtualSecureModeEnclaveProviderBase.cs

```diff
@@ -200,7 +204,13 @@ private X509Certificate2Collection GetSigningCertificate(string attestationUrl,
 
                 try
                 {
+#if NET9_0_OR_GREATER
```

Comment:
All these conditionals because of cert handling could be removed when System.Security.Cryptography.Pkcs is added to all target frameworks. It seems supported for net462/net8+ and would make it unified